packages/shared_preferences/shared_preferences_android/android/src/main/java/io/flutter/plugins/sharedpreferences/SharedPreferencesPlugin.java

```diff
@@ -5,42 +5,216 @@
+    TaskQueue taskQueue = messenger.makeBackgroundTaskQueue();
+    preferences = context.getSharedPreferences(SHARED_PREFERENCES_NAME, Context.MODE_PRIVATE);
+
+    try {
+      SharedPreferencesApi.setup(messenger, this);
+    } catch (Exception ex) {
+      Log.e(TAG, "Received exception while setting up SharedPreferencesPlugin", ex);
+    }
+  }
 
   @SuppressWarnings("deprecation")
   public static void registerWith(io.flutter.plugin.common.PluginRegistry.Registrar registrar) {
     final SharedPreferencesPlugin plugin = new SharedPreferencesPlugin();
-    plugin.setupChannel(registrar.messenger(), registrar.context());
+    plugin.setup(registrar.messenger(), registrar.context());
   }
 
   @Override
   public void onAttachedToEngine(FlutterPlugin.FlutterPluginBinding binding) {
-    setupChannel(binding.getBinaryMessenger(), binding.getApplicationContext());
+    setup(binding.getBinaryMessenger(), binding.getApplicationContext());
   }
 
   @Override
   public void onDetachedFromEngine(FlutterPlugin.FlutterPluginBinding binding) {
-    teardownChannel();
+    SharedPreferencesApi.setup(binding.getBinaryMessenger(), null);
+  }
+
+  @Override
+  public Boolean setBool(String key, Boolean value) {
+    return preferences.edit().putBoolean(key, value).commit();
+  }
+
+  @Override
+  public Boolean setString(String key, String value) {
+    if (value.startsWith(LIST_IDENTIFIER)
+        || value.startsWith(BIG_INTEGER_PREFIX)
+        || value.startsWith(DOUBLE_PREFIX)) {
+      throw new RuntimeException(
+          "StorageError: This string cannot be stored as it clashes with special identifier prefixes");
+    }
+    return preferences.edit().putString(key, value).commit();
+  }
+
+  @Override
+  public Boolean setInt(String key, Object value) {
+    Number number = (Number) value;
+    if (number instanceof BigInteger) {
+      BigInteger integerValue = (BigInteger) number;
+      return preferences
+          .edit()
+          .putString(key, BIG_INTEGER_PREFIX + integerValue.toString(Character.MAX_RADIX))
+          .commit();
+    } else {
+      return preferences.edit().putLong(key, number.longValue()).commit();
+    }
+  }
+
+  @Override
+  public Boolean setDouble(String key, Double value) {
+    String doubleValueStr = Double.toString(value);
+    return preferences.edit().putString(key, DOUBLE_PREFIX + doubleValueStr).commit();
+  }
+
+  @Override
+  public Boolean setStringList(String key, List<String> value) throws RuntimeException {
+    try {
+      Boolean success =
+          preferences.edit().putString(key, LIST_IDENTIFIER + encodeList(value)).commit();
+      return success;
+    } catch (RuntimeException e) {
+      throw e;
+    }
+  }
+
+  @Override
+  public Map<String, Object> getAll() throws RuntimeException {
+    try {
+      Map<String, Object> data = getAllPrefs();
+      return data;
+    } catch (RuntimeException e) {
+      throw e;
+    }
```

Comment:
Why are you catching exceptions everywhere just to rethrow them without changes?

Reply:
Sure I will ensure that I don't rethrow it.